airbyte-integrations/connectors/source-mysql-v2/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceOperations.kt

```diff
@@ -49,17 +58,65 @@ import io.airbyte.cdk.read.SelectNode
 import io.airbyte.cdk.read.SelectQuery
 import io.airbyte.cdk.read.SelectQueryGenerator
 import io.airbyte.cdk.read.SelectQuerySpec
+import io.airbyte.cdk.read.Stream
 import io.airbyte.cdk.read.Where
 import io.airbyte.cdk.read.WhereClauseLeafNode
 import io.airbyte.cdk.read.WhereClauseNode
 import io.airbyte.cdk.read.WhereNode
+import io.airbyte.cdk.read.cdc.DebeziumState
 import io.airbyte.cdk.util.Jsons
+import io.airbyte.integrations.source.mysql.cdc.MySqlDebeziumOperations
+import io.airbyte.integrations.source.mysql.cdc.MySqlPosition
 import io.micronaut.context.annotation.Primary
 import jakarta.inject.Singleton
+import java.time.OffsetDateTime
 
 @Singleton
 @Primary
-class MysqlSourceOperations : JdbcMetadataQuerier.FieldTypeMapper, SelectQueryGenerator {
+class MysqlSourceOperations :
+    JdbcMetadataQuerier.FieldTypeMapper, SelectQueryGenerator, JdbcAirbyteStreamFactory {
+
+    override val globalCursor: MetaField = MysqlCdcMetaFields.CDC_CURSOR
+
+    override val globalMetaFields: Set<MetaField> =
+        setOf(
+            MysqlCdcMetaFields.CDC_CURSOR,
+            CommonMetaField.CDC_UPDATED_AT,
+            CommonMetaField.CDC_DELETED_AT,
+            MysqlCdcMetaFields.CDC_LOG_FILE,
+            MysqlCdcMetaFields.CDC_LOG_POS
+        )
+
+    override fun decorateRecordData(
+        timestamp: OffsetDateTime,
+        globalStateValue: OpaqueStateValue?,
+        stream: Stream,
+        recordData: ObjectNode
+    ) {
+        recordData.set<JsonNode>(
+            CommonMetaField.CDC_UPDATED_AT.id,
+            CdcOffsetDateTimeMetaFieldType.jsonEncoder.encode(timestamp),
+        )
+        recordData.set<JsonNode>(
+            MysqlCdcMetaFields.CDC_LOG_POS.id,
+            CdcIntegerMetaFieldType.jsonEncoder.encode(0),
+        )
+        if (globalStateValue == null) {
```

Comment:
if it's not even a global state should we still decorate the record?

Reply:
As of now with the `GlobalLockResource` I don't think that that situation can even happen.